include/library/spdm_common_lib.h

```diff
@@ -232,16 +232,41 @@ void libspdm_set_last_spdm_error_struct(void *spdm_context,
                                         libspdm_error_struct_t *last_spdm_error);
 
 /**
- * Initialize an SPDM context.
+ * Initialize an SPDM context, as well as all secured message contexts,
+ * in the specified locations.
  *
- * The size in bytes of the spdm_context can be returned by libspdm_get_context_size.
+ * The size in bytes of the spdm_context can be returned by
+ * libspdm_get_context_size_without_secured_context.
  *
- * @param  spdm_context  A pointer to the SPDM context.
+ * The size in bytes of a single secured message context can be returned by
+ * libspdm_secured_message_get_context_size.
+ *
+ * @param  spdm_context          A pointer to the SPDM context.
+ * @param  secured_contexts      An array of pointers, with each entry containing
+ *                               the location of a secured message context.
+ * @param  num_secured_contexts  Number of secured message contexts to initialize.
+ *                               Currently, only LIBSPDM_MAX_SESSION_COUNT is supported.
+ *
+ * @retval RETURN_SUCCESS        Contexts are initialized.
+ * @retval RETURN_DEVICE_ERROR   Context initialization failed.
+ */
+libspdm_return_t libspdm_init_context_with_secured_context(void *context,
+                                                           void **secured_contexts,
+                                                           size_t num_secured_contexts);
+
+/**
+ * Initialize an SPDM context, as well as secured message contexts.
+ * The secured message contexts are appended to the context structure.
+ *
+ * The total size in bytes of the spdm_context and all secured message
+ * contexts can be returned by libspdm_get_context_size().
+ *
+ * @param  spdm_context         A pointer to the SPDM context.
  *
  * @retval RETURN_SUCCESS       context is initialized.
  * @retval RETURN_DEVICE_ERROR  context initialization failed.
  */
-libspdm_return_t libspdm_init_context(void *spdm_context);
+libspdm_return_t libspdm_init_context(void *context);
```

Comment:
`context` -> `spdm_context`

Reply:
I noticed that in a lot of these, the .h file has `void *spdm_context`, but the equivalent .c function has `void *context`, usually done as to avoid a naming conflict with a `libspdm_context_t spdm_context` variable.

Should we keep the current split, or clean this up for consistency between .c/.h files?

Reply:
The main requirement is that for publicly visible functions in the header the name is `spdm_context` so that it's not to be confused with any other context out there, and it's `void *` so the type is opaque. In the implementation we can do whatever is convenient. @jyao1 may have an opinion on the implementation side.

Reply:
Updated as requested